lighthouse/mcs-compliance.md

```diff
@@ -0,0 +1,176 @@
+Submariner to load balance this VIP to PodIPs in `EndpointSlices`. If VIP is not present, it will behave as it does
+today, i.e. return one of the ClusterIPs from `EndpointSlices`.
+
+### Conflict Resolution
+
+In case of conflict in VIP and PodIP settings when exporting `Service`, configuration on the oldest `ServiceExport`
+will be used. The `ServiceExport` in conflict will not be exported and `ServiceExportConflict` condition will be set.
+The `Valid` condition will also be set to `true` as Service itself is exportable.
+
+### Creating `ServiceExport`
+
+Here are the detailed steps when creating a `ServiceExport`:
+
+1. When a new `ServiceExport` is created, check if `ServiceImport` exists or not.
+2. If `ServiceImport` doesn't exist, allocate the VIP and add it to new `ServiceImport`.
+3. If `ServiceImport` exists, compare the combination of annotations on `ServiceExport` and global flags against
+annotations in `ServiceImport` for any conflicts.
+4. In case of conflict, set `ServiceExportConflict` condition on the `ServiceExport` and attempt conflict resolution.
+5. In case conflict can't be resolved, don't export the service.
+6. If PodIP annotation is set, create `EndpointSlices` with PodIPs instead of ClusterIP. Follow existing rules for
+merging port information in `EndpointSlices`.
+
+### Updating annotation on ServiceExport
+
+Changing annotation will not modify the current behavior of using VIP or PodIPs. It will only change the conflict
+condition on `ServiceExport` if it adds or resolves the conflict, depending on the change. If the users wish to change
+the behavior for a given `Service`, they will need to delete all `ServiceExports` and recreate them with required
+annotations.
+
+### Deleting `ServiceExport`
+
+Here are the detailed steps when deleting a `ServiceExport`:
+
+1. Check the cluster list on `ServiceImport`. If this is the last cluster to Export the service. If not, proceed as
+usual irrespective of VIP and PodIP annotations. This means deleting EndpointSlice, recalculating service ports and
+updating cluster list on `ServiceImport`.
+2. If this is last cluster, deallocate the VIP if it is same CIDR as current cluster and delete the `ServiceImport`.
+
+### Globalnet
+
+Globalnet will not be supported yet. Supporting Globalnet with PodIPs will be a significant scale issue as each Pod
+backing the Service will require a GlobalIngressIP.
+
+For any use cases that require Globalnet support with this feature, a separate Enhancement can be proposed in the
+future.
+
+### Migration
+
+Since there is no change in default behavior, nothing should be required for migration.
+
+If users want to add annotations on existing ServiceExports post deployment, they will need to delete ServiceExports.
+
+## Design Details
+
+### Global Flags for deploy broker
+
+When deploying broker, `--enable-vip` and `--endpointslices-with-pod-ips` flags will be added. There will also be an
```

Comment:
```suggestion
When deploying broker, `--enable-clusterset-vip` and `--endpointslices-with-pod-ips` flags will be added. There will also be an
```

Reply:
I didn't use `clusterset` coz at broker level it is implied. Wanted to keep the keyword short.

Reply:
I think we should be consistent with our terminology everywhere.